pkg/image/api/helper.go

```diff
@@ -290,6 +371,30 @@ func DockerImageReferenceForStream(stream *ImageStream) (DockerImageReference, e
 	return ParseDockerImageReference(spec)
 }
 
+// FollowTagReference walks through the defined tags on a stream, following any referential tags in the stream.
+// Will return ok if the tag is valid, multiple if the tag had at least reference, and ref and finalTag will be the last
+// tag seen. If a circular loop is found ok will be false.
+func FollowTagReference(stream *ImageStream, tag string) (finalTag string, ref *TagReference, ok bool, multiple bool) {
+	seen := map[string]struct{}{}
+	for {
+		tagRef, ok := stream.Spec.Tags[tag]
```

Comment:
Specs define references, not status.  Status is the observed tagged images.  Specs define intent.